### PR TITLE
KEYCLOAK-19056 EDIT MODE field should not be leave empty

### PR DESCRIPTION
**Pull request by mposolda**

### CODE REVIEW

**Comment by mposolda** [comment]
@pskopek Do you please have a chance to review this?

**Comment by mposolda** [comment]
@pskopek @iankko I fixed the test failures in this PR. Does anyone of you has a chance to take a look at this? Thanks

**Review by pskopek — APPROVED** [review]
All looks good!

**Review by iankko — APPROVED** [review]
TL; DR: LGTM AFAICT (see below for a more detailed review)

The difference in behavior of the `distribution` build from [Keycloak v15.0.2](https://github.com/keycloak/keycloak/releases/download/15.0.2/keycloak-15.0.2.zip) release and the `distribution` build built from this PR is clearly visible, and the one from this PR looks more reasonable to me:

Using the embedded ApacheDS LDAP server via `-Pldap` profile:

- With the `15.0.2` release, the `Edit Mode` allows to be left to the default (unset / empty / blank) value. Saving the provider, syncing the users. Trying to edit some bits of a particular user throws:

`org.keycloak.storage.ReadOnlyException: Federated storage is not writable`

error only later.

- With `distribution` build from this PR, the `Edit Mode` is mandatory already (it's not possible to save the provider definition without choosing a value). Syncing the users, and subsequent editing works as expected AFAICT (throws the same error when `READ_ONLY` mode was chosen, succeeds when `WRITABLE` mode was chosen).

ACK

**Comment by mposolda** [comment]
@pskopek @iankko Thanks!

Added also pipeline build with LDAP and admin console tests: https://keycloak-jenkins.com/job/universal-test-pipeline-server/775

**Comment by molequetu** [comment]
Hello, sorry for the interruption in the current issue, but it was the one I found with reference to 'Federated storage is not writable'

I am trying to override and add globalization to this message returned from KC when federation storage is in READONLY mode and forget password funcionality from user action is used.

I was searching on theme login messages but didn't find any related message, e.g https://github.com/keycloak/keycloak/blob/master/themes/src/main/resources/theme/base/login/messages/messages_en.properties

Is there any way to change this message and add globalized message for this key too;
